packages/core/parcel-bundler/src/builtins/hmr-runtime.js

```diff
@@ -20,29 +20,32 @@ function Module(moduleName) {
 }
 
 module.bundle.Module = Module;
-var checkedAssets, assetsToAccept;
+var assetsToCheck, checkedAssets, assetsToAccept;
 
 var parent = module.bundle.parent;
 if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
   var hostname = process.env.HMR_HOSTNAME || location.hostname;
   var protocol = location.protocol === 'https:' ? 'wss' : 'ws';
   var ws = new WebSocket(protocol + '://' + hostname + ':' + process.env.HMR_PORT + '/');
   ws.onmessage = function(event) {
+    assetsToCheck = [];
     checkedAssets = {};
     assetsToAccept = [];
 
     var data = JSON.parse(event.data);
 
     if (data.type === 'update') {
-      var handled = false;
       data.assets.forEach(function(asset) {
-        if (!asset.isNew) {
-          var didAccept = hmrAcceptCheck(global.parcelRequire, asset.id);
-          if (didAccept) {
-            handled = true;
-          }
-        }
+        if (!asset.isNew) assetsToCheck.push(asset.id);
       });
+      var handled = false;
+      var id;
+      while (id = assetsToCheck.shift()) {
```

Comment:
Could you change this to something like

```JavaScript
while (id) {
  ...
  id = assetsToCheck.shift()
}
```

Looks a little less complex.
Everything else looks fine.

Reply:
Good point - fixed.